python/dgl/graphbolt/graph_storage/sampled_sub_graph.py

```diff
@@ -0,0 +1,156 @@
+"""Heterogeneous sampled subgraph."""
+# pylint: disable= invalid-name
+from collections.abc import Mapping
+
+
+class SampledSubGraph:
+    r"""Class for sampled subgraph."""
+
+    def __init__(
+        self,
+        node_pairs,
+        reverse_column_node_ids=None,
+        reverse_row_node_ids=None,
+        reverse_edge_ids=None,
+    ):
+        """ ""Initialize the SampledSubGraph object. In the context of a
+        heterogeneous graph, each field should be of `Dict` type. For
+        homogeneous graphs, each field should correspond to its respective
+        value type."
+
+        Parameters
+        ----------
+        node_pairs :  Tuple[Tensor] or Dict[(str, str, str), Tuple[Tensor]]
```

Comment:
Do we prefer COO here or CSC?

Reply:
Both can be supported in this structure, because both `COO` and  `CSC` are 2 tensors. In our canonical case, the subgraph isn't compacted. Therefore, we opt for `COO` over `CSC` to avoid numerous gaps that would occur with `CSC`.